core/store/tx_manager.go

```diff
@@ -39,8 +41,38 @@ const (
 	nonceReloadBackoffBaseTime = 3 * time.Second
 )
 
-// ErrPendingConnection is the error returned if TxManager is not connected.
-var ErrPendingConnection = errors.New("Cannot talk to chain, pending connection")
+var (
+	// ErrPendingConnection is the error returned if TxManager is not connected.
+	ErrPendingConnection = errors.New("Cannot talk to chain, pending connection")
+
+	promNumGasBumps = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "tx_manager_num_gas_bumps",
+		Help: "Number of gas bumps per transaction",
+	},
+		[]string{"tx_hash", "attempt_index"},
+	)
+
+	promGasBumpExceedsLimit = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "tx_manager_gas_bump_exceeds_limit",
+		Help: "Number of times gas bumping failed from exceeding the configured limit. Any counts of this type indicate a serious problem.",
+	},
+		[]string{"tx_hash", "attempt_index"},
+	)
+
+	promGasBumpUnderpricedReplacement = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "tx_manager_gas_bump_underpriced_replacement",
+		Help: "Number of underpriced replacement errors received while trying to bump gas. Counts of this type most likely indicate some kind of misconfiguration or problem.",
+	},
+		[]string{"tx_hash", "attempt_index"},
+	)
+
+	promTxAttemptFailed = promauto.NewCounterVec(prometheus.CounterOpts{
```

Comment:
Is CounterVec the right one to use?

Reply:
I think you just want counter.

Reply:
I want to include the tx_hash and attempt_index though...